.github/workflows/generate-config.yaml

```diff
@@ -0,0 +1,133 @@
+          if [ "${{ github.event.pull_request.merged }}" != "true" ]; then
+            echo "PR was not merged. Exiting."
+            exit 0
+          fi
+
+      - name: Get the merge commit SHA
+        id: merge_commit
+        run: |
+          MERGE_COMMIT=$(git log -1 --pretty=format:"%H")
+          echo "MERGE_COMMIT=$MERGE_COMMIT" >> $GITHUB_ENV
+
+      - name: Determine NETWORKS
+        id: determine_networks
+        run: |
+          echo "Checking modified files in commit ${{ env.MERGE_COMMIT }}"
+          MODIFIED_FILES=$(git diff --name-only HEAD~1..HEAD)
+          echo "Modified files: $MODIFIED_FILES"
+
+          NETWORKS=()
+
+          if echo "$MODIFIED_FILES" | grep -q "axelar-chains-config/info/mainnet.json"; then
+            NETWORKS+=("mainnet")
+          fi
+          if echo "$MODIFIED_FILES" | grep -q "axelar-chains-config/info/testnet.json"; then
+            NETWORKS+=("testnet")
+          fi
+          if echo "$MODIFIED_FILES" | grep -q "axelar-chains-config/info/stagenet.json"; then
+            NETWORKS+=("stagenet")
+          fi
+          if echo "$MODIFIED_FILES" | grep -q "axelar-chains-config/info/devnet-amplifier.json"; then
+            NETWORKS+=("devnet-amplifier")
+          fi
+
+          if [ ${#NETWORKS[@]} -eq 0 ]; then
+            echo "No relevant changes detected."
+            echo "NETWORKS=none" >> $GITHUB_ENV
+            exit 0
+          fi
+
+          echo "Detected networks: ${NETWORKS[*]}"
+          echo "NETWORKS=${NETWORKS[*]}" >> $GITHUB_ENV
+
+      - name: Checkout external repository
+        if: env.NETWORKS != 'none'
+        uses: actions/checkout@v4
+        with:
+          repository: axelarnetwork/${{ secrets.EXTERNAL_REPO }}
+          path: ${{ secrets.EXTERNAL_REPO }}
+          token: ${{ secrets.CHAINS_PAT }}
+
+      - name: Install jq
+        if: env.NETWORKS != 'none'
+        run: sudo apt-get install -y jq
+
+      - name: Setup Branch Name
+        if: env.NETWORKS != 'none'
+        id: branch-name
+        run: echo "branch=feat/config-update-$(date +'%Y-%m-%d-%H-%M-%S')" >> "$GITHUB_OUTPUT"
+
+      - name: Create branch and generate config
+        if: env.NETWORKS != 'none'
+        run: |
+          cd ${{ secrets.EXTERNAL_REPO }}
+          curl -sSL https://install.python-poetry.org | python3 -
+          poetry install --no-interaction
+          git config --local user.email "devops@axelar.network"
+          git config --local user.name "devops"
+          git checkout -b ${{ steps.branch-name.outputs.branch }}
+
+          for NETWORK in ${{ env.NETWORKS }}; do
+            echo "Generating config for $NETWORK"
+            poetry run python generate.py --network "$NETWORK" gen-configs all-configs
+          done
+
+          git add --all
+          git commit -m "create branch ${{ steps.branch-name.outputs.branch }}"
+          git push --set-upstream origin ${{ steps.branch-name.outputs.branch }}
+
+      - name: Create Pull Request
```

Comment:
why not just commit directly instead of creating a PR and then merging it?

Reply:
@milapsheth It's essentially for organization purposes, conflicts detection, and having better historical context and branch protection in case it's set, since having the action committing directly to the `main` branch with no history of "when and what" will make things not clear and the changes will be lost among the commits history.

plus, any changes will be in another branch it will be in multiple commits, so merging directly the branch to the `main` one doesn't always have to go well in case some conflicts happen and in addition committing directly to the `main` branch isn't a best practice, always PRs need to take place to have better insights what's being changed/pushed.

Reply:
correct if the user of the PAT has higher privileges